vignettes/sliding.Rmd

```diff
@@ -0,0 +1,281 @@
+
+In this vignette, we demonstrate using `epi_slide()` to slide ARX forecasters over an `epi_df` object and compare the results obtained from using different forecasting engines. We then compare the results from version-aware and unaware forecasting, where former is obtained from applying `epix_slide()` to the `epi_archive` object, while the latter is from applying `epi_slide()` to the latest snapshot of the data.
+
+## Comparing different forecasting engines
+
+### Example using CLI and case data from CA and FL 
+
+First, we download the version history (ie. archive) of the percentage of doctor’s visits with CLI (COVID-like illness) computed from medical insurance claims and the number of new confirmed COVID-19 cases per 100,000 population (daily) for CA and FL from the COVIDcast API. We process as before, with the modification that we use `sync = locf` in `epix_merge()` so that the last version of each observation can be carried forward to extrapolate unavailable versions for the less up-to-date input archive.
+
+```{r grab-epi-data}
+theme_set(theme_bw())
+#%% Remove below covidcast code and library(covidcast) 
+# y <- covidcast_signals(
+#   c("doctor-visits", "jhu-csse"),
+#   c("smoothed_adj_cli", "confirmed_7dav_incidence_prop"),
+#   start_day = "2020-06-01", 
+#   end_day = "2021-12-01",
+#   issues = c("2020-06-01", "2021-12-01"),
+#   geo_type = "state", 
+#   geo_values = c("ca", "fl")) 
+# saveRDS(y, "/inst/extdata/ca_fl_covidcast_signals.rds")
+y <- readRDS(
+  system.file("extdata", "ca_fl_covidcast_signals.rds", package = "epipredict", mustWork = TRUE)
+)
+ 
+y1 <- y[[1]] %>% 
+  select(geo_value, time_value, version = issue, percent_cli = value) %>%
+  as_epi_archive()
+
+x <- epix_merge(
+  y1, y[[2]] %>% 
+    select(geo_value, time_value, version = issue, case_rate = value) %>%
+    as_epi_archive(), sync = "locf")
+```
+
+After obtaining the latest snapshot of the data, we produce forecasts on that data using the default engine of simple linear regression and compare to a random forest (as were introduced in the Getting started with epipredict vignette). #%% Add link to getting started vignette once html file is there!
+
+Note that all of the warnings about `forecast_date` being less than the most recent update date of the data have been suppressed to avoid cluttering the output.
+
+```{r make-arx-kweek, warning = FALSE}
+# Latest snapshot of data, and forecast dates
+x_latest <- epix_as_of(x, max_version = max(x$versions_end)) 
+fc_time_values <- seq(as.Date("2020-08-01"), as.Date("2021-12-01"), 
+                      by = "1 month")
+
+k_week_ahead <- function(ahead = 7, engine_type = "linear_reg") {
+  if(engine_type == "linear_reg"){
+    x_latest %>%
+      epi_slide(function(x, ...) 
```

Comment:
Does this work with 

```r
x_latest %>% 
  epi_slide(~arx_forecaster(.x, outcome=...)
```

Reply:
Yes. Sliding with a formula or function format works... I'll use the formula format if that's preferred.

Reply:
Yeah, It looks a bit cleaner.